sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -146,6 +162,22 @@ class SQLConfSuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("reset specific property - internal conf") {
+    spark.sessionState.conf.clear()
+    val original = spark.conf.get(SQLConf.OPTIMIZER_MAX_ITERATIONS)
+    try {
+      assert(spark.conf.get(SQLConf.OPTIMIZER_MAX_ITERATIONS) === 100)
+      sql(s"set ${SQLConf.OPTIMIZER_MAX_ITERATIONS.key}=10")
+      assert(spark.conf.get(SQLConf.OPTIMIZER_MAX_ITERATIONS) === 10)
+      assert(sql(s"set").where(s"key = '${SQLConf.OPTIMIZER_MAX_ITERATIONS.key}'").count() == 1)
+      sql(s"reset ${SQLConf.OPTIMIZER_MAX_ITERATIONS.key}")
+      assert(spark.conf.get(SQLConf.OPTIMIZER_MAX_ITERATIONS) === 100)
+      assert(sql(s"set").where(s"key = '${SQLConf.OPTIMIZER_MAX_ITERATIONS.key}'").count() == 0)
+    } finally {
+      sql(s"set ${SQLConf.OPTIMIZER_MAX_ITERATIONS}=$original")
+    }
+  }
```

Comment:
After done this, building threw `SQLConfSuite is not a test`, maybe this style does not work. @gatorsmile